scalding-core/src/main/scala/com/twitter/scalding/ArgHelp.scala

```diff
@@ -0,0 +1,106 @@
+package com.twitter.scalding
+
+sealed abstract class DescribedArg {
+  def key: String
+  def description: String
+}
+
+case class RequiredArg(key: String, description: String) extends DescribedArg
+case class OptionalArg(key: String, description: String) extends DescribedArg
+case class ListArg(key: String, description: String) extends DescribedArg
+case class BooleanArg(key: String, description: String) extends DescribedArg
+
+trait ArgHelper {
+  /**
+   * Similar to describe but validate all args are described
+   * @param describedArgs List of Argument Descriptions
+   * @param ex Input Execution
+   * @return Output Execution
+   */
+  def validatedDescribe(describedArgs: Seq[DescribedArg], ex: Execution[Unit]): Execution[Unit] = {
```

Comment:
Mmm I think maybe an exception is the way to go here. I think this API will be a bit funky otherwise. Since if you continue on from the returned execution from this in a chain it will throw away the fact the help was printed since its not represented in the type system that you should exit here. This would wind up really only being suitable for the root entry point from Execution App? It seems like you should be able to zip two of these together or flatMap them. 

@johnynek ^ thoughts?